Winget-AutoUpdate/Winget-Upgrade.ps1

```diff
@@ -14,27 +14,76 @@ $Script:IsSystem = [System.Security.Principal.WindowsIdentity]::GetCurrent().IsS
 #Run log initialisation function
 Start-Init
 
-#Log running context
+#Get WAU Configurations
+$Script:WAUConfig = Get-ItemProperty -Path "HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\Winget-AutoUpdate"
+
+#Log running context and more...
 if ($IsSystem) {
     Write-Log "Running in System context"
-}
-else {
-    Write-Log "Running in User context"
-}
 
-#Get WAU Configurations
-$Script:WAUConfig = Get-ItemProperty -Path "HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\Winget-AutoUpdate"
+    #Get and set Domain/Local Policies (GPO)
+    $ActivateGPOManagement, $ChangedSettings = Get-Policies
+    if ($ActivateGPOManagement) {
+        Write-Log "Activated WAU GPO Management detected, comparing..."
+        if ($null -ne $ChangedSettings -and $ChangedSettings -ne 0) {
+            Write-Log "Changed settings detected and applied" "Yellow"
+        }
+        else {
+            Write-Log "No Changed settings detected" "Yellow"
+        }
+    }
 
-#Run post update actions if necessary
-if (!($WAUConfig.WAU_PostUpdateActions -eq 0)) {
-    Invoke-PostUpdateActions
-}
+    # Maximum number of log files to keep. Default is 3. Setting MaxLogFiles to 0 will keep all log files.
+    $MaxLogFiles = $WAUConfig.WAU_MaxLogFiles
+    if ($null -eq $MaxLogFiles) {
+        [int32] $MaxLogFiles = 3
+    }
+    else {
+        [int32] $MaxLogFiles = $MaxLogFiles
+    }
+    
+    # Maximum size of log file.
+    $MaxLogSize = $WAUConfig.WAU_MaxLogSize
+    if (!$MaxLogSize) {
+        [int64] $MaxLogSize = 1048576 # in bytes, default is 1048576 = 1 MB
+    }
+    else {
+        [int64] $MaxLogSize = $MaxLogSize
+    }
 
-#Run Scope Machine funtion if run as system
-if ($IsSystem) {
+    #LogRotation if System
+    $Exception, $Rotate = Invoke-LogRotation $LogFile $MaxLogFiles $MaxLogSize
+    if ($Exception -eq $True) {
+        Write-Log "An Exception occured during Log Rotation..."
+    }
+    if ($Rotate -eq $True) {
+        #Log Header
+        $Log = "##################################################`n#     CHECK FOR APP UPDATES - $(Get-Date -Format (Get-culture).DateTimeFormat.ShortDatePattern)`n##################################################"
+        $Log | out-file -filepath $LogFile -Append
+        Write-Log "Running in System context"
+        if ($ActivateGPOManagement) {
+            Write-Log "Activated WAU GPO Management detected, comparing..."
+            if ($null -ne $ChangedSettings -and $ChangedSettings -ne 0) {
+                Write-Log "Changed settings detected and applied" "Yellow"
+            }
+            else {
+                Write-Log "No Changed settings detected" "Yellow"
+            }
+        }
+        Write-Log "Max Log Size reached: $MaxLogSize bytes - Rotated Logs"
+    }
```

Comment:
Quick question, from line 56 to 74, could this not be part of Invoke-LogRotation itself ? to get Winget-Upgrade.ps1 "cleaner" ?

Reply:
Possibly...

Reply:
haha. just asking, not an obligation

Reply:
Do you want me to examine it (and possibly change)?

Reply:
only if you have time for it. don't spend time for nothing ^^

Reply:
#270